src/libcore/time.rs

```diff
@@ -130,10 +143,12 @@ impl Duration {
     /// ```
     #[stable(feature = "duration", since = "1.3.0")]
     #[inline]
-    #[rustc_const_stable(feature = "duration_consts", since = "1.32.0")]
-    pub fn new(secs: u64, nanos: u32) -> Duration {
-        let secs =
-            secs.checked_add((nanos / NANOS_PER_SEC) as u64).expect("overflow in Duration::new");
+    #[rustc_const_unstable(feature = "const_checked_int_methods", issue = "53718")]
```

Comment:
```suggestion
    #[rustc_const_unstable(feature = "const_duration_new", issue = "53718")]
```
I don't think we should reuse `const_checked_int_methods` here.
You might have to add `#![feature(const_duration_new)]` to src/libcore/lib.rs so `Duration::MAX` can use this.

Reply:
Also there should be a new tracking issue then.